### PR TITLE
Release v0.5.0

### PR DESCRIPTION
**Pull request by relyt29**
## Summary

1. Implement a fix for incomplete migration of topic fields from the v0.4.0 upgrade. 
2. Add additional RPC endpoint to allow for recalculating the inflation rate / target emission rate more often than once per month.
3. Update to use rewardable topics as active topics instead of using them separately.
4. Other miscellaneous minor fixes

## Link(s) to Ticket(s) or Issue(s) resolved by this PR

See [Changelog.md](https://github.com/allora-network/allora-chain/blob/release-v0.5.0/CHANGELOG.md)

### CODE REVIEW

File: .github/workflows/build_push_upgrader_docker_hub.yml [hunk 1]
```diff
@@ -11,11 +11,11 @@ on:
       allorad_current_version:
         description: 'Current version of allorad'
         required: true
-        default: 'v0.3.0'
+        default: 'v0.4.0'
       allorad_upgrade_version:
         description: 'Upgrade version of allorad'
         required: true
-        default: 'v0.4.0'
+        default: 'v0.4.1'
```
Comment (fernandofcampos) [inline .github/workflows/build_push_upgrader_docker_hub.yml:18]: needs to be v0.5.0

File: Dockerfile.cosmovisor [hunk 2]
```diff
@@ -38,8 +38,8 @@ ADD https://github.com/cosmos/cosmos-sdk/releases/download/cosmovisor/${COSMOVIS
 RUN tar -xvf /tmp/cosmovisor.tar.gz -C /usr/local/bin/ && \
     chmod +x /usr/local/bin/cosmovisor
 
-ARG ALLORAD_CURRENT_VERSION="v0.3.0"
-ARG ALLORAD_UPGRADE_VERSION="v0.4.0"
+ARG ALLORAD_CURRENT_VERSION="v0.4.0"
+ARG ALLORAD_UPGRADE_VERSION="v0.4.1"
```
Comment (fernandofcampos) [inline Dockerfile.cosmovisor:42]: needs to be v0.5.0

File: Dockerfile.upgrade [hunk 3]
```diff
@@ -31,7 +31,7 @@ RUN --mount=type=cache,target=/gocache \
     --mount=type=cache,target=/root/.cache/go-build \
     make build-local-edits
 
-RUN cp build/allorad /go/bin/allorad-v0.4.0
+RUN cp build/allorad /go/bin/allorad-v0.4.1
```
Comment (fernandofcampos) [inline Dockerfile.upgrade:34]: needs to be v0.5.0

File: app/upgrades/v0_4_1/upgrades.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"context"
+
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	"github.com/allora-network/allora-chain/app/upgrades"
+	"github.com/cosmos/cosmos-sdk/types/module"
+)
+
+const (
+	UpgradeName = "v0.4.1"
```
Comment (fernandofcampos) [inline app/upgrades/v0_4_1/upgrades.go:12]: needs to be v0.5.0. 

The folder also needs to be renamed.

File: docker-compose.yaml [hunk 5]
```diff
@@ -2,14 +2,14 @@
 services:
   node:
     container_name: sample_validator
-    image: "alloranetwork/allora-chain:v0.4.0-upgrader" # docker image without cosmovisor. use vx.x.x-upgrader for upgrade image
+    image: "alloranetwork/allora-chain:v0.4.1-docker-upgrade" # docker image without cosmovisor. use vx.x.x-docker-upgrade for upgrade image
```
Comment (fernandofcampos) [inline docker-compose.yaml:5]: needs to be v0.5.0

File: scripts/upgrade.sh [hunk 6]
```diff
@@ -5,8 +5,8 @@
 # to the network to trigger the upgrade process.
 
 # Update the variables below to match current conditions and desired upgrade
-current_verison=v0.3.0
-new_version=v0.4.0
+current_verison=v0.4.0
+new_version=v0.4.1
```
Comment (fernandofcampos) [inline scripts/upgrade.sh:9]: needs to be v0.5.0

File: test/integration/upgrade_test.go [hunk 7]
```diff
@@ -51,10 +51,10 @@ func voteOnProposal(m testCommon.TestConfig, proposalId uint64) {
 	}
 }
 
-// propose an upgrade to the v0.4.0 software version
+// propose an upgrade to the v0.4.1 software version
```
Comment (fernandofcampos) [inline test/integration/upgrade_test.go:54]: needs to be v0.5.0

File: test/integration/upgrade_test.go [hunk 8]
```diff
@@ -51,10 +51,10 @@ func voteOnProposal(m testCommon.TestConfig, proposalId uint64) {
...
 func proposeUpgrade(m testCommon.TestConfig) (proposalId uint64, proposalHeight int64) {
 	ctx := context.Background()
-	name := v0_4_0.UpgradeName
+	name := v0_4_1.UpgradeName
```
Comment (fernandofcampos) [inline test/integration/upgrade_test.go:57]: needs to be v0.5.0

File: test/local_testnet_upgrade_l1.sh [hunk 9]
```diff
@@ -30,7 +30,7 @@ VALIDATORS_API_PORT_START=1317
 HEADS_IP_START=20
 CHAIN_ID="${CHAIN_ID:-devnet}"
 LOCALNET_DATADIR="$(pwd)/$CHAIN_ID"
-UPGRADE_VERSION="v0.4.0"
+UPGRADE_VERSION="v0.4.1"
```
Comment (fernandofcampos) [inline test/local_testnet_upgrade_l1.sh:33]: needs to be v0.5.0

**Review by fernandofcampos — DISMISSED** [review]

File: .github/workflows/build_push_upgrader_docker_hub.yml (hunk 1 above)
Comment (relyt29) [reply .github/workflows/build_push_upgrader_docker_hub.yml:18]: https://github.com/allora-network/allora-chain/pull/594

File: Dockerfile.cosmovisor (hunk 2 above)
Comment (relyt29) [reply Dockerfile.cosmovisor:42]: https://github.com/allora-network/allora-chain/pull/594

File: Dockerfile.upgrade (hunk 3 above)
Comment (relyt29) [reply Dockerfile.upgrade:34]: https://github.com/allora-network/allora-chain/pull/594

File: app/upgrades/v0_4_1/upgrades.go (hunk 4 above)
Comment (relyt29) [reply app/upgrades/v0_4_1/upgrades.go:12]: https://github.com/allora-network/allora-chain/pull/594

File: docker-compose.yaml (hunk 5 above)
Comment (relyt29) [reply docker-compose.yaml:5]: https://github.com/allora-network/allora-chain/pull/594

File: scripts/upgrade.sh (hunk 6 above)
Comment (relyt29) [reply scripts/upgrade.sh:9]: https://github.com/allora-network/allora-chain/pull/594

File: test/integration/upgrade_test.go (hunk 8 above)
Comment (relyt29) [reply test/integration/upgrade_test.go:57]: https://github.com/allora-network/allora-chain/pull/594

File: test/local_testnet_upgrade_l1.sh (hunk 9 above)
Comment (relyt29) [reply test/local_testnet_upgrade_l1.sh:33]: https://github.com/allora-network/allora-chain/pull/594

**Review by kpeluso — APPROVED** [review]

**Pull request merged**